apps/src/code-studio/components/progress/LessonProgress.jsx

```diff
@@ -161,11 +162,31 @@ class LessonProgress extends Component {
     return {headerFullProgressOffset: 0, vignetteStyle: null};
   }
 
+  isBonusComplete() {
+    return !!this.props.levels.find(
```

Comment:
This could also be written as `return this.props.levels.some(level => level.bonus && level.status === LevelStatus.perfect)`

Reply:
👍